core/lib/dal/src/transactions_dal.rs

```diff
@@ -1293,7 +1304,27 @@ impl TransactionsDal<'_, '_> {
         }
     }
 
-    pub async fn get_call_trace(&mut self, tx_hash: H256) -> sqlx::Result<Option<Call>> {
+    pub async fn get_call_trace(&mut self, tx_hash: H256) -> anyhow::Result<Option<Call>> {
+        let protocol_version: Option<ProtocolVersionId> = sqlx::query!(
+            r#"
+            SELECT
+                protocol_version
+            FROM
+                miniblocks
+                LEFT JOIN transactions ON transactions.miniblock_number = miniblocks.number
+            WHERE
+                transactions.hash = $1
```

Comment:
This query is formally correct but I'm not sure how postgres will execute it. Instead I suggest to do
`FROM transactions LEFT JOIN miniblocks ON transactions.miniblock_number = miniblocks.number`, this way postgres will 100% execute the query optimally.